### PR TITLE
Tabs Toggle Feature for the overflowed tabs

### LINKED ISSUE #4379: Scroll feature to see multiple circuites
### Is your feature request related to a problem? Please describe

Right now after using the whole simulator width, if we even want to create circuits it limits the width to that particular screen size

Screenshot 2023-12-15 152553
 

### Describe the solution you'd like

Instead of that, we can let the user to _**scroll**_ through the circuits

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by Malavi1**
Fixes #4379

#### Describe the changes you have made in this PR -
* introduced caretup & caretdown toggle icons on the extreme right hand of the tabs bar
* Implemented a conditional on click function 
* Changes the CSS accordingly 

### Screenshots of the changes (If any) -

https://github.com/CircuitVerse/CircuitVerse/assets/112646623/61189b3d-115e-44a8-a2d0-90a586009b20

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by Malavi1** [comment]
@tanmoysrt can you help me with this issue!

**Comment by tanmoysrt** [comment]
> @tanmoysrt can you help me with this issue!

It seems that the build phase failed. Try to run this locally to track the error
```
yarn build
```

File: simulator/src/circuit.js [hunk 1]
```diff
@@ -570,4 +575,4 @@ export default class Scope {
             }
         }
     }
-}
+
```
Comment (tanmoysrt) [inline simulator/src/circuit.js:578]: @Malavi1 check this one, maybe the closing bracket got missed

**Comment by Malavi1** [comment]
@tanmoysrt it is good to go IG

File: app/views/simulator/edit.html.erb [hunk 2]
```diff
@@ -817,6 +818,30 @@
...
+        });
+    } else if (tabsCaret.hasClass('fa-solid') && tabsCaret.hasClass('fa-square-caret-up')) {
+        tabsCaret.removeClass('fa-solid fa-square-caret-up');
+        tabsCaret.addClass('fa-solid fa-square-caret-down');
+        tabsBar.css({
+            'height': 'auto',
+            'overflow': 'visible'
+        });
+    }
+}
```
Comment (tanmoysrt) [inline app/views/simulator/edit.html.erb:843]: It will be better to write this codes in vanilla js instead of jquery

**Review by tanmoysrt — CHANGES_REQUESTED** [review]

**Comment by tachyons** [comment]
Since the focus is now on vue-js version, could you move the changes to the vue-js version ?

**Comment by tanmoysrt** [comment]
> Since the focus is now on vue-js version, could you move the changes to the vue-js version ?

There is 

> Since the focus is now on vue-js version, could you move the changes to the vue-js version ?

In `cv-frontend-vue` these changes are made by another contributor Ref- https://github.com/CircuitVerse/cv-frontend-vue/issues/222

File: app/views/simulator/edit.html.erb (hunk 2 above)
Comment (Malavi1) [reply app/views/simulator/edit.html.erb:843]: ok, I will do that but may I know the reason particularly not using jquery?!

**Comment by Malavi1** [comment]
@tanmoysrt can look at this once!

**Comment by tanmoysrt** [comment]
> @tanmoysrt can look at this once!

You can ignore the CI issue. It's not related to your changes

File: simulator/src/circuit.js [hunk 3]
```diff
@@ -171,9 +171,14 @@ export function newCircuit(name, id, isVerilog = false, isVerilogMain = false) {
...
         } else {
-            var html = `<div style='' class='circuits toolbarButton current' draggable='true' id='${scope.id}'><span class='circuitName noSelect'>${truncateString(name, 18)}</span><span class ='tabsCloseButton' id='${scope.id}'  >x</span></div>`;
+            var html = `<div style='' class='circuits toolbarButton current' draggable='true' id='${scope.id}'><span class='circuitName noSelect'>${truncateString(name, 18)}</span><span class ='tabsCloseButton' id='${scope.id}'>x</span></div>`;
             $('#tabsBar').children().last().before(html);
         }
+        // var addBtn = "<button name='logixButton' class='logixButton' id='createNewCircuitScope' onclick='' aria-label='logix button'><i class='fas fa-plus'></i></button>";
+        // $('#tabsBar').append(addBtn);
+
+        // var addBtn = "<button name='logixButton' class='logixButton' id='createNewCircuitScope' onclick='' aria-label='logix button'><i class='fas fa-plus'></i></button>";
+        // $('#tabsBar').children().last().before(html);
```
Comment (tanmoysrt) [inline simulator/src/circuit.js:181]:
```suggestion
```

Remove comments

**Review by tanmoysrt — APPROVED** [review]
Working as expected.

cc @Prerna-0202 @Arnabdaz

**Pull request closed without merging**